scRNA-seq-advanced/setup/intro-diff-expression-setup.Rmd

```diff
@@ -0,0 +1,624 @@
+dge_list
+```
+
+Now we need to normalize these counts! 
+
+Use the trimmed mean of M-values method to compute normalization factors. 
+Counts are now large enough to apply bulk normalization methods rather than the deconvolution method that was used on individual libraries. 
+
+```{r}
+# compute some normalization factors 
+dge_list <- edgeR::calcNormFactors(dge_list)
+# should now see a new column, `norm.factors`
+dge_list$samples
+```
+
+Before we do any DGE, let's do some QC checks to make sure our data looks okay. 
+We make some Mean difference plots, but I'm not really convinced we get a whole lot from these? 
+
+```{r}
+# mean difference plots
+purrr::walk(1:ncol(dge_list), 
+            ~ limma::plotMD(dge_list, column = .x))
+```
+
+```{r}
+# MDS scaling plot 
+# first compute counts per million 
+counts_per_million <- edgeR::cpm(dge_list, log = TRUE)
+
+dge_list$samples <- dge_list$samples %>%
+  dplyr::mutate(diagnosis_groups = dplyr::case_when(
+    subdiagnosis == "Alveolar rhabdomyosarcoma" ~ "ARMS",
+    subdiagnosis == "Embryonal rhabdomyosarcoma" ~ "ERMS"
+  ))
+
+limma::plotMDS(counts_per_million,
+        # add some colors based on our two groups
+        col = ifelse(dge_list$samples$diagnosis_groups == "ARMS", "black", "red"))
+```
+
+Let's do some DE now! 
+
+```{r}
+# set up design matrix to compare between ARMS and ERMS groups
+design_matrix <- model.matrix(~factor(subdiagnosis),
+                              dge_list$samples)
+design_matrix
+```
+
+Estimate dispersions using both the quasi-likelihood(QL) dispersion and the negative binomial(NB) dispersions. 
+
+```{r}
+# estimate NB dispersions 
+dge_list <- edgeR::estimateDisp(dge_list, design = design_matrix)
+summary(dge_list$trended.dispersion)
+```
+
+```{r}
+edgeR::plotBCV(dge_list)
+```
+
+```{r}
+# estimate the quasi-likelihood dispersions
+glm_fit <- edgeR::glmQLFit(dge_list, design_matrix, robust = TRUE)
+edgeR::plotQLDisp(glm_fit)
+```
+
+Test for differences using the `glmQLFTest` where differentially expressed gens are defined as those with non-zero LFC and a FDR of < 5%.
+
+```{r}
+results <- edgeR::glmQLFTest(glm_fit, coef = ncol(design_matrix))
+summary(limma::decideTests(results))
+```
+```{r}
+edger_results <- edgeR::topTags(results, n="Inf")$table
+edger_results_sig <- edger_results %>%
+  dplyr::filter(FDR <= 0.05)
+
+# print out results 
+edger_results_sig
+```
+
+### Perform differential expression with DESeq2
+
+Let's do the same thing, but try with DESeq2.
+
+```{r}
+# deseq2 requires the coldata object as a separate input
+coldata_df <- as.data.frame(colData(tumor_myoblast_sce)) %>%
+  dplyr::mutate(diagnosis_groups = dplyr::case_when(
+    subdiagnosis == "Alveolar rhabdomyosarcoma" ~ "ARMS",
+    subdiagnosis == "Embryonal rhabdomyosarcoma" ~ "ERMS"
+  ),
+  diagnosis_groups = as.factor(diagnosis_groups)) 
+
+# set the levels
+coldata_df$diagnosis_groups <- relevel(coldata_df$diagnosis_groups, ref = "ERMS")
```

Comment:
I think you need to do this also for edgeR (or remove it so ARMS is always first level), since based on the ggplot compared their significance in the last section the results are opposite sign. So, we want to make sure the direction of comparison is the same if we want to use both DGE methods.